tests/MongoDB.Driver.Core.Tests/Core/Compression/CompressorsTests.cs

```diff
@@ -30,6 +32,91 @@ public class CompressorsTests
+
+            string TestCase(string incomeMessage, string expectedCompressed, string expectedDecompressed)
+            {
+                var bytes = Encoding.ASCII.GetBytes(incomeMessage);
+                var compressor = GetCompressor(compressorType);
+
+                var inputStream = GetStreamFromBytes(bytes);
+                var outpustStream = GetStreamFromBytes();
+
+                compressor.Compress(inputStream, outpustStream);
+
+                var compressed = GetBytesFromStream(outpustStream);
+                var compressedStr = string.Join(";", compressed);
+                compressedStr.Should().Be(expectedCompressed);
+
+                var compressedBytes = GetBytesFromStream(outpustStream);
+                var list = new List<byte>();
+                list.AddRange(new byte[]
+                {
+                    // Filled here: https://github.com/robertvazan/snappy.net/blob/master/Snappy.NET/SnappyFrame.cs#L342
+                    255, 6, 0, 0,   // `255` - shows the type for next header section. 255 - `StreamIdentifier`, the next 6 bytes will parsed as `StreamIdentifier` header.
+                    // Contains `ASCII` bytes from `sNaPpY` string. So, it's constant bytes.
+                    //https://github.com/robertvazan/snappy.net/blob/master/Snappy.NET/SnappyFrame.cs#L26
+                    115, 78, 97, 80, 112, 89, // contant bytes
+                    // NOTE: the above header bytes are constants!!
+
+
+
+                    // Contains information about compressed data
+                    // show the type of data after the headers. `0` - Compressed: https://github.com/robertvazan/snappy.net/blob/master/Snappy.NET/SnappyFrameType.cs#L16.
+                    // It means that next block(after the checksum header) will be tried to be decompressed.
+                    // `47` - the size of all content after the current header. Which includes a `checksum` header + compressed data.
+                    0, 47, 0, 0,  // so the full size of compressed data is `43` (47 - 4)
+
+                    33, 133, 232, 85 // The checkSum header. Calculated here: https://github.com/robertvazan/snappy.net/blob/master/Snappy.NET/SnappyFrame.cs#L493
+                });
+                list.AddRange(compressedBytes.ToList());
+                compressedBytes = list.ToArray();
+                var outStream = GetStreamFromBytes(compressedBytes);
+                var decStream = GetStreamFromBytes();
+                outpustStream.Position = 0;
+                compressor.Decompress(outStream, decStream);
```

Comment:
When I manually restored the expected `snappy` headers, decompression passed without any problems